node.h

```diff
@@ -0,0 +1,29 @@
+#ifndef NODE_H
+#define NODE_H
+
+class Node
+{
+    public:
+    int data;
```

Comment:
Isn't it better to make the member variable data private so that its value can't be changed from the outside. Also the name data isn't  very informative about the content that's going to be stored in the variable.